modules/snooker/src/main/ts/ephox/snooker/model/Transitions.ts

```diff
@@ -38,10 +38,16 @@ const toGrid = (warehouse: Warehouse, generators: Generators, isNew: boolean): S
   const grid: Structs.RowCells[] = [];
 
   Arr.each(warehouse.colgroups, (colgroup) => {
-    const cols = Arr.map(colgroup.columns, (column): Structs.ElementNew =>
-      Structs.elementnew(column.element, isNew, false)
-    );
-    grid.push(Structs.rowcells(colgroup.element, cols, 'colgroup', isNew));
+    const colgroupCols: Structs.ElementNew[] = [];
+    // This will add missing cols as well as clamp the number of cols to the max number of actual columns
+    // Note: Spans on cols are unsupported so clamping cols may result in a span on a col element being incorrect
+    for (let columnIndex = 0; columnIndex < warehouse.grid.columns; columnIndex++) {
+      const element = Warehouse.getColumnAt(warehouse, columnIndex)
+        .map((column) => Structs.elementnew(column.element, isNew, false))
+        .getOrThunk(() => Structs.elementnew(generators.colGap(), true, false));
```

Comment:
This is probably just me not understanding table code, but where in this do the new `col` elements actually enter the dom?

Reply:
I believe they are entered when the `Redraw.render` function is called after a table operation is complete